code/index.d.ts

```diff
@@ -1,26 +1,28 @@
-// Type definitions for code 4.0.0
+// Type definitions for code 4.0
 // Project: https://github.com/hapijs/code
 // Definitions by: Prashant Tiwari <https://github.com/prashaantt>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /** Generates an assertion object. */
-export function expect<T>(value: T | T[], prefix?: string): AssertionChain<T>;
+declare function expect<T>(value: T | T[], prefix?: string): AssertionChain<T>;
```

Comment:
Since the types are the public API, they should be available publicly. Try compiling this code with the `--declaration` option:
```ts
interface Foo { x: number; }
export function foo(): Foo { return { x: 0 }; }
```
It gives you `error TS4060: Return type of exported function has or is using private name 'Foo'.`.
Even if you try and keep the return type private, it can be worked around by calling `const fnRet = fn()` and using `typeof fnRet` instead of `Int`.